csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -121,13 +112,14 @@ void exitCallback(int ret, string msg, bool silent)
                 throw;
             }
 
-            RestoreNugetPackages(allNonBinaryFiles, allProjects, allSolutions, dllLocations);
+            var dllLocations = fileProvider.Dlls.Select(x => new AssemblyLookupLocation(x)).ToHashSet();
```

Comment:
Are we sure that none of the preceding code adds any dlls to the sourcedir?

Reply:
That's a good question. In the second commit I made the `FileProvider.all` not lazy. So the question is if there's any code that adds DLLs before the `new FileProvider(...)` call. And I think the answer is no.